### PR TITLE
consistent energy fitting

### PR DESCRIPTION
**Pull request by njzjz**

### CODE REVIEW

File: source/tests/consistent/fitting/test_ener.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+    @property
+    def skip_pt(self) -> bool:
+        (
+            resnet_dt,
+            precision,
+            distinguish_types,
+            numb_fparam,
+        ) = self.param
+        # TODO: float32 has bug
```
Comment (njzjz) [inline source/tests/consistent/fitting/test_ener.py:83]: @anyangml It should be fixed

File: deepmd/pt/model/model/__init__.py [hunk 2]
```diff
@@ -54,6 +54,7 @@ def get_zbl_model(model_params):
     fitting_net["ntypes"] = descriptor.get_ntypes()
     fitting_net["distinguish_types"] = descriptor.distinguish_types()
     fitting_net["embedding_width"] = descriptor.get_dim_out()
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/pt/model/model/__init__.py:56]: It's still used by the dipole fitting.

[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply deepmd/pt/model/model/__init__.py:56]: @anyangml please not here we should unify the name of the parameter. `embedding_width`  is prefered to `dim_rot_mat`.

**Review by wanghan-iapcm — APPROVED** [review]

File: source/tests/consistent/fitting/test_ener.py (hunk 1 above)
Comment (anyangml) [reply source/tests/consistent/fitting/test_ener.py:83]: what's the issue here?

**Review by anyangml — APPROVED** [review]

File: source/tests/consistent/fitting/test_ener.py (hunk 1 above)
Comment (wanghan-iapcm) [reply source/tests/consistent/fitting/test_ener.py:83]: it seems that float 32 does not pass the consistency test. what is the difference? do we reduce the tolerance accordingly when using float32?

Comment (njzjz) [reply source/tests/consistent/fitting/test_ener.py:83]: > do we reduce the tolerance accordingly when using float32?

It just complains that a float64 tensor cannot be operated with a float32 tensor.

Comment (wanghan-iapcm) [reply source/tests/consistent/fitting/test_ener.py:83]: please raise an issue for it. Thanks!

Comment (njzjz) [reply source/tests/consistent/fitting/test_ener.py:83]: I found only pt has the bug. The dp backend had the problem of tolerance, and I fixed that.

File: source/tests/consistent/fitting/test_ener.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        (
+            resnet_dt,
+            precision,
+            distinguish_types,
+            numb_fparam,
+        ) = self.param
+        if precision == "float64":
+            return 1e-10
+        elif precision == "float32":
+            return 1e-5
```
Comment (njzjz) [inline source/tests/consistent/fitting/test_ener.py:183]: It looks `1e-5` may still fail the test...
```suggestion
            return 1e-4
```

**Pull request merged**